core/src/main/java/org/eclipse/ec4j/core/EditorConfigSession.java

```diff
@@ -169,12 +180,13 @@ public EditorConfigLoader getLoader() {
      *
      * @param resource
      *            the resource to query the {@link Property}s for
-     * @return an immutable {@link Collection} of {@link Property}s applicable to the given {@link Resource}
-     * @throws EditorConfigException
+     * @return a {@link QueryResult} that contains {@link Property}s applicable to the given {@link Resource}
+     * @throws IOException
+     *             on I/O problems during the reading from the given {@link Resource}
      */
-    public Collection<Property> queryProperties(Resource resource) throws EditorConfigException {
-        Map<String, Property> oldProperties = Collections.emptyMap();
-        Map<String, Property> properties = new LinkedHashMap<>();
+    public QueryResult queryProperties(Resource resource) throws IOException {
+        QueryResult.Builder result = QueryResult.builder();
+        ArrayList<EditorConfig> editorConfigs = new ArrayList<>();
```

Comment:
Should be better to write List<EditorConfig> instead of ArrayList<EditorConfig>.

Reply:
Yes, that's a leftover. I originally wanted to use `editorConfigs.listIterator(editorConfigs.size())` to be able to iterate in the reverse order, but then decided to access by index https://github.com/angelozerr/ec4j/pull/57/files/c09486350b57597f3a6b4f9e0c57199c34612e64#diff-3de86c371b365c772dec5b132b0b954dR205 

But thanks for noting this, I actually thought `java.util.List` does not have `listIterator(int index)` which I now see it has. So I should have used `List` for the local var anyway :)